server/src/repositories/media.repository.ts

```diff
@@ -156,10 +169,29 @@ export class MediaRepository implements IMediaRepository {
   }
 
   private configureFfmpegCall(input: string, output: string | Writable, options: TranscodeCommand) {
+    let lastProgressFrame: number = 0;
+    const { frameCount, percentInterval } = options.progress;
+    const frameInterval = Math.ceil(frameCount / (100 / percentInterval));
+    const isDebug = this.logger.isLevelEnabled(LogLevel.DEBUG);
     return ffmpeg(input, { niceness: 10 })
       .inputOptions(options.inputOptions)
       .outputOptions(options.outputOptions)
       .output(output)
-      .on('error', (error, stdout, stderr) => this.logger.error(stderr || error));
+      .on('start', (command: string) => this.logger.debug(command))
+      .on('error', (error, _, stderr) => this.logger.error(stderr || error))
+      .on('progress', (progress: ProgressEvent) => {
+        if (!isDebug || !frameCount || !frameInterval || progress.frames - lastProgressFrame < frameInterval) {
```

Comment:
It would probably be cleaner to not even add the event handler in the first place, rather than add it and skip a bunch of values. You are using the current log level to determine and optimize the implementation and the ffmpeg library might do the same (not send progress events if there are no subscribers).

I would either say add it dynamically or always log the debug log and let the logger do it's job and filter the values out.

Reply:
Oh true, that's a good point.